src/xGadgetron/cGadgetron/gadgetron_data_containers.cpp

```diff
@@ -188,79 +186,95 @@ int
+        }
+        num_acq++;
     }
 
     int const num_dims = 3;
     dim[0] = ns;
     dim[1] = nc;
-    dim[2] = na;
+    dim[2] = num_acq;
 
     return num_dims;
 }
 
 uint16_t MRAcquisitionData::get_trajectory_dimensions(void) const
 {
     int na = number();
-    ASSERT(na>0, "You are asking for dimensions on an empty acquisition container. Please dont... ");
+    ASSERT(na > 0, "You are asking for dimensions on an empty acquisition container. Please don't...");
 
     ISMRMRD::Acquisition acq;
-    get_acquisition(0,acq);
-
-    uint16_t const traj_dims = acq.trajectory_dimensions();
-    bool trajectory_consistent = true;
-    for(int i=1; i<na; ++i)
+    uint16_t traj_dims;
+    int num_acq = 0;
+    for (int i = 0; i < na; ++i)
     {
         get_acquisition(i, acq);
-        trajectory_consistent *= (traj_dims== acq.trajectory_dimensions());
+        if (TO_BE_IGNORED(acq))
+            continue;
+        if (num_acq == 0)
+            traj_dims = acq.trajectory_dimensions();
+        else if (acq.trajectory_dimensions() != traj_dims)
+            throw LocalisedException("Not every acquisition in your container has the same trajectory dimension." , __FILE__, __LINE__);
+        num_acq++;
     }
-    if(trajectory_consistent)
-        return traj_dims;
-    else
-        throw LocalisedException("Not every acquisition in your container has the same trajectory dimension." , __FILE__, __LINE__);
+    return traj_dims;
 }
 
 void MRAcquisitionData::get_kspace_dimensions(std::vector<size_t>& dims) const
 {
     int na = number();
-    ASSERT(na>0, "You are asking for dimensions on an empty acquisition container. Please dont... ");
+    ASSERT(na > 0, "You are asking for dimensions on an empty acquisition container. Please don't...");
 
     ISMRMRD::Acquisition acq;
-    get_acquisition(0,acq);
-
-    int nro = acq.number_of_samples();
-    int nc = acq.active_channels();
 
-    std::vector<size_t> empty_data;
-    dims.swap(empty_data);
+    int nro;
+    int nc;
 
-    for(int i=1; i<na; ++i)
+    int num_acq = 0;
+    for (int i = 0; i < na; ++i)
     {
         get_acquisition(i, acq);
-
-        if(acq.active_channels() !=nc)
-            throw std::runtime_error("The number of channels is not consistent within this container.");
-        if(acq.number_of_samples() != nro)
-            throw std::runtime_error("The number of readout points is not consistent within this container.");
+        if (TO_BE_IGNORED(acq))
+            continue;
+        if (num_acq == 0) {
+            nro = acq.number_of_samples();
+            nc = acq.active_channels();
+        }
+        else {
+            if (acq.active_channels() != nc)
+                throw std::runtime_error("The number of channels is not consistent within this container.");
+            if (acq.number_of_samples() != nro)
+                throw std::runtime_error("The number of readout points is not consistent within this container.");
+        }
+        num_acq++;
```

Comment:
Same here the `num_acq` variable is never used again but still always increased.